tools/swiftdt/Sources/swiftdt/main.swift

```diff
@@ -0,0 +1,122 @@
+import ArgumentParser
+import SwiftRemoteMirror
+
+
+func argFail(_ message: String) -> Never {
+  print(message, to: &Std.err)
+  exit(EX_USAGE)
+}
+
+func machErrStr(_ kr: kern_return_t) -> String {
+  let errStr = String(cString: mach_error_string(kr))
+  let errHex = String(kr, radix: 16)
+  return "\(errStr) (0x\(errHex))"
+}
+
+func dumpConformanceCache(context: SwiftReflectionContextRef) throws {
+  try context.iterateConformanceCache { type, proto in
+    let typeName = context.name(metadata: type) ?? "<unknown>"
+    let protoName = context.name(proto: proto) ?? "<unknown>"
+    print("Conformance: \(typeName): \(protoName)")
+  }
+}
+
+func dumpMetadataAllocations(context: SwiftReflectionContextRef) throws {
+  var allocations = context.allocations
+  for allocation in allocations {
+    print("Metadata allocation at: \(hex: allocation.ptr) " +
+          "size: \(allocation.size) tag: \(allocation.tag)")
+  }
+  
+  allocations.sort()
+  let metadatas = allocations.compactMap { $0.metadata(in: context) }
+
+  for metadata in metadatas {
+    print("Metadata \(hex: metadata.ptr)")
+    print("    Name: \(metadata.name)")
+
+    if let allocation = allocations.last(where: { metadata.ptr >= $0.ptr }) {
+      let offset = metadata.ptr - allocation.ptr
+      print("    In allocation \(hex: allocation.ptr) " +
+            "size \(allocation.size) at offset \(offset)")
+    } else {
+      print("    Not in any known metadata allocation. How strange.")
+    }
+  }
+}
+
+func makeReflectionContext(
+  nameOrPid: String
+) -> (Inspector, SwiftReflectionContextRef) {
+  guard let pid = pidFromHint(nameOrPid) else {
+    argFail("Cannot find pid/process \(nameOrPid)")
```

Comment:
If you don’t mind making this a throwing method, you can use `ValidationError` here.
```suggestion
    throw ValidationError("Cannot find pid/process \(nameOrPid)")
```